### ISSUE TITLE
Getting location updates only via GPS, but not via Network Provider

### ISSUE THREAD

**Issue by 2SI3NX** [tananaev/traccar-client-android#247]
So im successfully getting updates via GPS Provider or Mixed Provider when location is turned on.
But if location is turned off, im assuming im still supposed to get updates via Mixed Provider or Network Provider.. but im not.

Android v6.0.1

**Comment by tananaev** [comment]
If location it turned off completely, you won't get any location obviously. Not sure why you assume that you should get something.

**Comment by 2SI3NX** [comment]
If thats the case, then what does Network Provider mean exactly? it means it uses cellphone towers to triangulate location, regardless whether GPS location is turned on or off, isnt that so?
If not, then what is the purpose of Network Provider option if it wont work or use cellphone towers?

**Comment by tananaev** [comment]
Did you turn GPS off or all location providers off? Can I see a screenshot of the location settings screen?

**Comment by 2SI3NX** [comment]
![selection_008](https://cloud.githubusercontent.com/assets/10450920/22400173/7deff4be-e5b6-11e6-8fc8-179134acf595.jpg)
thats the only thing i turned off, only GPS.  and ive even enabled Data.
Surely it should still try and detect through the network(cell towers) ?
Other apps, such as wechat for example would still use location accurately using AGPS while location is turned off as in the screenshot attached.

**Comment by tananaev** [comment]
That's the thing that turns off all locations, including network.

**Comment by 2SI3NX** [comment]
yes but, other apps **somehow** seem to use cell provider location or cell towers **even if that location setting is turned off.**  I dont know how they do it, but maybe the option of 'Network Provider' (in Android Traccar Client) can be enhanced or improved even if Location in Android settings is off?  Is this possible?

**Comment by tananaev** [comment]
Maybe it could use raw cell and wifi data, but that's not how it currently works. There is a network location provider in Android and that's what the app uses.

**Comment by 2SI3NX** [comment]
ok. Would you add this as an enhancement to traccar client? It would be very useful without having location settings turned on all the time, which compromises the privacy of the user with the involvement of the rest of the apps that the user doesnt want location on for.

**Comment by tananaev** [comment]
I will consider implementing it when I have time.

**Comment by glbatistabh** [comment]
I have the same problem. When I turn off the GPS, the APP stops sending the positions. And it is very common, the user to leave the GPS turned off.
Are you forecast if you will implement this feature?

**Comment by hoaadv** [comment]
I have another problem but I think it similar with this one.
When the cell phone loss GPS signal, It's sent message "Connectivity change" to server. And nothing shows in server likely there is no connection from client to server.

I think the best way should be: client sent message "Lost Signal" or something like that then server show the "Lost signal" status.

Thank you!

**Comment by tananaev** [comment]
Connectivity change has nothing to do with GPS. It is printed when network connection status changes. It's not possible to send anything to the server if there is no connection.

**Comment by joelmanx** [comment]
it will can add the feature by to obtain the GPS Location of way hidden?, I am programmer and I want to contribute at the project

**Comment by tananaev** [comment]
You are welcome to contribute, but I don't understand your question.
